SimpleNotePad.java

```diff
@@ -0,0 +1,437 @@
+
+				out.close();
+
+			} catch (IOException ex) {
+
+
+
+			}
+
+		}else if(event.getActionCommand().equals("print")) { 	
+
+			try{
+
+				PrinterJob pjob = PrinterJob.getPrinterJob();
+
+				pjob.setJobName("Sample Command Pattern");
+
+				pjob.setCopies(1);
+
+				pjob.setPrintable(new Printable() {
+
+
+
+					public int print(Graphics pg, PageFormat pf, int pageNum) {
+
+						if (pageNum>0)		
+
+							return Printable.NO_SUCH_PAGE;	
+
+
+
+						pg.drawString(noteArea.getText(), 500, 500);	
+
+						paint(pg);
+
+
+
+						return Printable.PAGE_EXISTS;
+
+					}
+
+				});
+
+
+
+				if (pjob.printDialog() == false)	
+
+					return;				
+
+
+
+				pjob.print();			 
+
+			} catch (PrinterException pe) { 
+
+				JOptionPane.showMessageDialog(null,
+
+						"Printer error" + pe, "Printing error", 
+
+						JOptionPane.ERROR_MESSAGE);
+		}
+			}	
+
+	
+
+		else if(event.getActionCommand().equals("replace")) {
+			//pasteImplementation();
+			
+			String find = noteArea.getSelectedText();
+			String replace = "";
+
+			if(!find.isEmpty()) {
+			replace = JOptionPane.showInputDialog("Replace or insert with");
+			int n = JOptionPane.showConfirmDialog(noteArea,"Are you sure you want to replace?",
+			"View carefully", JOptionPane.YES_NO_OPTION);	
+
+			if (n == JOptionPane.YES_OPTION) {
+			noteArea.replaceSelection(replace);
+			} else if (n == JOptionPane.NO_OPTION) {
+			return;
+			} 
+
+		}}
+
+		else if(event.getActionCommand().equals("paste")) { 
+
+			pasteImplementation();
+
+		}else if(event.getActionCommand().equals("ok")) { 
+
+			String text=noteArea.getSelectedText();
+			String replace1=textInput.getText();
+			
+			if(text!=null){
+				replace1=textInput.getText().replaceAll(text, replace1);
+				noteArea.setText(replace1);
+			}
+
+		}
+
+		else if(event.getActionCommand().equals("Open File")) {
+
+			JFileChooser open = new JFileChooser();  
+
+			int option = open.showOpenDialog(this);
+
+			if (option == JFileChooser.APPROVE_OPTION) {
+
+				File file = null;
+
+				FileReader filer = null;
+
+				try {
+
+					file = open.getSelectedFile();
+
+					String filepath = file.getAbsolutePath();
+
+					noteArea.setText(filepath);
+
+					filer = new FileReader(file);
+
+					JMenuItem newItem = new JMenuItem(filepath);
+
+					newItem.setActionCommand(filepath);
+
+					newItem.addActionListener(this);
+
+					rFile.add(newItem);
+
+					openRecent.add(newItem);
+
+					noteArea.read(filer,"");
+
+				} catch (Exception ex) {
+
+					System.out.println(ex.getMessage());
+
+				}
+
+			}
+
+		}
+
+		else {
+
+			for(JMenuItem recent : rFile) {
+
+				if(event.getActionCommand().equals(recent.getText())) {
+
+					FileReader filer = null;
+
+					File file = null;
+
+					try {
+
+						file = new File(event.getActionCommand());
+
+						filer = new FileReader(file);
+
+						noteArea.read(filer, "");
+
+					}
+
+					catch(Exception ex) {
+
+						System.out.println(ex.getMessage());
+
+					}
+
+				}
+
+			}

```

Comment:
This class is very long. It may be easier to read if it is split up into different classes.